File: baekjoon/silver/1340/1340.py
# input
# January 31, 1900 00:47
# output
# 8.228120243531203

month = {
    'January': 1,
    'February': 2,
    'March': 3,
    'April': 4,
    'May': 5,
    'June':6,
    'July':7,
    'August':8,
    'September':9,
    'October':10,
    'November':11,
    'December':12
}

def is_leap_yaer(year: int) -> bool:
  return year % 400 == 0 or (year % 4 == 0 and year % 100 != 0)

def get_total_time(year: int) -> int:
  return 366 if is_leap_yaer(year) else 365

leap_year = [0, 31, 29, 31, 30, 31, 30, 31, 31, 30, 31 ,30, 31]
non_leap_year = [0, 31, 28, 31, 30, 31, 30, 31, 31, 30, 31 ,30, 31]

for i in range(1, 13):
  leap_year[i] += leap_year[i - 1]
  non_leap_year[i] += non_leap_year[i - 1]

arr = input().split()
y, m, d, h, mm = map(int, [arr[2], month[arr[0]], arr[1][:-1]] + arr[3].split(':'))

total = get_total_time(y) * 24 * 60
now = (((leap_year[m - 1] if is_leap_yaer(y) else non_leap_year[m - 1]) + d - 1) * 24 + h) * 60 + mm
print(now / total * 100)

File: baekjoon/silver/1340/1340.go
package main

import "fmt"

func main() {
	var month map[string]int = map[string]int{
		"January":   1,
		"February":  2,
		"March":     3,
		"April":     4,
		"May":       5,
		"June":      6,
		"July":      7,
		"August":    8,
		"September": 9,
		"October":   10,
		"November":  11,
		"December":  12,
	}
	var leapYear []int = []int{0, 31, 29, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31}
	var nonLeapYear []int = []int{0, 31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31}
	var y, m, d, h, mm, now int
	var ms string
	for i := 1; i < 13; i++ {
		leapYear[i] += leapYear[i-1]
		nonLeapYear[i] += nonLeapYear[i-1]
	}
	fmt.Scanf("%s %d, %d %d:%d", &ms, &d, &y, &h, &mm)
	m = month[ms]

	totalTime := getTotalDay(y) * 24 * 60
	if isLeapYear(y) {
		now = leapYear[m-1]*24*60 + (d-1)*24*60 + h*60 + mm
	} else {
		now = nonLeapYear[m-1]*24*60 + (d-1)*24*60 + h*60 + mm
	}

	fmt.Println(float64(now) / float64(totalTime) * 100)
}

func isLeapYear(year int) bool {
	return year%400 == 0 || (year%4 == 0 && year%100 != 0)
}

func getTotalDay(year int) int {
	if isLeapYear(year) {
		return 366
	}
	return 365
}
